system_tests/pubsub.py

```diff
@@ -169,7 +169,12 @@ def _by_timestamp(message):
         self.assertEqual(message2.data, MESSAGE_2)
         self.assertEqual(message2.attributes['extra'], EXTRA_2)
 
+    def _maybe_emulator_skip(self):
+        if os.getenv(PUBSUB_EMULATOR) is not None:
+            self.skipTest('IAM not supported by Pub/Sub emulator')
+
```

Comment:
The decorator gets created at import time, whereas these tests don't check state until runtime, so I was trying to stay true to that.